awswrangler/_distributed.py

```diff
@@ -99,6 +103,7 @@ def dispatch_on_engine(cls, func: FunctionType) -> FunctionType:
 
         @wraps(func)
         def wrapper(*args: Any, **kw: Dict[str, Any]) -> Any:
+            cls.initialize(name=cls.get().value)
```

Comment:
Can we check if `cls._initialized_engine` is set and run initialise only if it is not? I understand it's probably not an expensive operation right now but still seems a bit of a waste to run this every time we call a function

Reply:
It's already done within the initialize ray function itself